cmd/apprepository-controller/main.go

```diff
@@ -87,18 +155,19 @@ func main() {
 	}
 }
 
-func init() {
-	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	flag.StringVar(&repoSyncImage, "repo-sync-image", "quay.io/helmpack/chart-repo:latest", "container repo/image to use in CronJobs")
-	flag.StringVar(&repoSyncCommand, "repo-sync-cmd", "/chart-repo", "command used to sync/delete repos for repo-sync-image")
-	flag.StringVar(&namespace, "namespace", "kubeapps", "Namespace to discover AppRepository resources")
-	flag.BoolVar(&reposPerNamespace, "repos-per-namespace", true, "Defaults to watch for repos in all namespaces. Switch to false to watch only the configured namespace.")
-	flag.StringVar(&dbURL, "database-url", "localhost", "Database URL")
-	flag.StringVar(&dbUser, "database-user", "root", "Database user")
-	flag.StringVar(&dbName, "database-name", "charts", "Database name")
-	flag.StringVar(&dbSecretName, "database-secret-name", "kubeapps-db", "Kubernetes secret name for database credentials")
-	flag.StringVar(&dbSecretKey, "database-secret-key", "postgresql-root-password", "Kubernetes secret key used for database credentials")
-	flag.StringVar(&userAgentComment, "user-agent-comment", "", "UserAgent comment used during outbound requests")
-	flag.StringVar(&crontab, "crontab", "*/10 * * * *", "CronTab to specify schedule")
+// getImagePullSecretsRefs gets the []string of Secrets names from the
+// StringSliceVar flag list passed in the repoSyncImagePullSecrets arg
+func getImagePullSecretsRefs(imagePullSecretsRefsArr []string) []corev1.LocalObjectReference {
+	var imagePullSecretsRefs []corev1.LocalObjectReference
+
+	// if no repoSyncImagePullSecrets arg passed, return nil, as usual
+	if len(imagePullSecretsRefsArr) == 0 {
+		return nil
+	}
```

Comment:
This `if` is not actually needed, if `imagePullSecretsRefsArr` is empty, the `range` will be skipped and it will return `nil` anyway.